pkg/R/sparkR.R

```diff
@@ -23,30 +23,39 @@ connExists <- function(env) {
 
 # Stop the Spark context.
 # Also terminates the backend this R session is connected to
-sparkR.stop <- function(env) {
-  cat("Stopping SparkR\n")
+sparkR.stop <- function(env = .sparkREnv) {
 
   if (!connExists(env)) {
     # When the workspace is saved in R, the connections are closed
     # *before* the finalizer is run. In these cases, we reconnect
     # to the backend, so we can shut it down.
-    connectBackend("localhost", .sparkREnv$sparkRBackendPort)
+    tryCatch({
+      connectBackend("localhost", .sparkREnv$sparkRBackendPort)
+    },error = function(err){
+      cat("Error in Connection: Use sparkR.init() to Restart SparkR\n")
+    },warning = function(war){
+      cat("No Connection Found: Use sparkR.init() to Restart SparkR\n")
+    })
```

Comment:
Ah I see - Thanks. In that case can you move the `if (exists(".sparkRCon"))` above the calls to stopping the spark context ? If we don't have a connection we can't stop the spark context as well.